flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractionUtils.java

```diff
@@ -215,22 +215,33 @@ public static Type extractTypeArgument(Type t, int index) throws InvalidTypesExc
 	 * Extracts a Single Abstract Method (SAM) as defined in Java Specification (4.3.2. The Class Object,
 	 * 9.8 Functional Interfaces, 9.4.3 Interface Method Body) from given class.
 	 *
-	 * @param baseClass
-	 * @throws InvalidTypesException if the given class does not implement
-	 * @return
+	 * @param baseClass a class that is a FunctionalInterface to retrieve a SAM from
+	 * @throws InvalidTypesException if the given class does not implement FunctionalInterface
+	 * @return single abstract method of the given class
 	 */
 	public static Method getSingleAbstractMethod(Class<?> baseClass) {
+
+		if (!baseClass.isInterface()) {
+			throw new InvalidTypesException("FunctionalInterface must be an interface");
```

Comment:
Yes and no, I would say. Given class must be a FunctionalInterface which implies it has to be an interface. If you feel. How about such a message: "Given class must be a FunctionalInterface which implies it has to be an interface."?